### ISSUE TITLE
find a way to emulate RETURNING

### ISSUE THREAD

**Issue by mcr** [EnterpriseDB/mysql_fdw#95]
PG::InternalError:` ERROR:  RETURNING is not supported by this FDW : 
    INSERT INTO "crm_accounts" ("date_entered", "date_modified", "id") VALUES ($1, $2, $3) RETURNING "id"
    /sg1/web/beaumont/beaumont/shared/bundle/ruby/2.1.0/gems/activerecord-4.2.5/

I got this while running a rails migration against my test system which actually has our PG->Mysql FDW installed.  Sigh. Of course, Mysql doesn't support RETURNING to get access to the sequence numbers, but rather has the per-connection SELECT LAST_INSERT_ID();.  Could the FDW possibly emulate that? It would, I realize, be hard to do, since RETURNING in it's full generality can return multiple values, etc.

**Comment by ahsanhadi** [comment]
Yes you are right, mysql doesn't have support for returning clause. Not sure how we can emulate that with mysql_fdw. This is a feature request that we can look at once it is prioritised by our product management.

**Comment by KES777** [comment]
Maybe as workaround, may you passthrough 'SELECT LAST_INSERT_ID' queries from PostgreSQL to MySQL. Or do some special query that mysql_fdw will recognize and do 'SELECT LAST_INSERTPID' internally. For example the mysql_fdw may implement and supply UDF's with same name

**Comment by mcr** [comment]
I'm willing to hack the rails adapter to do something interesting as a work around.  I'm not sure exactly what you have in mind at the rails side. It seems like we could add the SELECT LAST_INSERT_ID to the sql we send to mysql, which is what I will look into doing.

**Comment by drm** [comment]
I ran into this today, using PostgREST on top of a MySQL foreign table. I have created a small patch, that turns the error into a WARNING, just to see what would break. Curiously enough, this doesn't break at all, all simple RETURNING clauses I tried (INSERT, UPDATE, DELETE) work as expected... 

So I think this definitely deserves some reconsideration. I will create a PR concerning this.

**Comment by drm** [comment]
I'm planning on working around the INSERT issues with a trigger managed in Postgres.

**Comment by mcr** [comment]
The ability to return the new primary key is used extensively by web frameworks in their postgresql drivers.

**Comment by drm** [comment]
Yeah that makes total sense, but there's an easy workaround by simply implementing a sequence in a trigger. With the error in place, there is no possible workaround.

**Comment by bartlaarhoven** [comment]
+1 for this issue, but in our case, the external database (MySQL / MariaDB) is not under our own control. So implementing a trigger isn't that obvious or even possible to do.
Note that MariaDB 10.5 and up does [support RETURNING in the way PostgreSQL does](https://mariadb.com/kb/en/insertreturning/). Maybe support can at least be built-in for underlying databases that support it?

**Comment by drm** [comment]
In my proposed solution the trigger is implemented in postgres, not MySQL.
